lib/db.c

```diff
@@ -1598,10 +1598,12 @@ grn_table_create_validate(grn_ctx *ctx,
     return ctx->rc;
   }
 
-  if ((flags & GRN_OBJ_KEY_LARGE) && table_type != GRN_OBJ_TABLE_HASH_KEY) {
+  if ((flags & GRN_OBJ_KEY_LARGE) && (table_type != GRN_OBJ_TABLE_HASH_KEY &&
+                                      table_type != GRN_OBJ_TABLE_PAT_KEY)) {
```

Comment:
```suggestion
  if ((flags & GRN_OBJ_KEY_LARGE) && table_type != GRN_OBJ_TABLE_HASH_KEY &&
                                      table_type != GRN_OBJ_TABLE_PAT_KEY) {
```

No need for brackets `()`?

Reply:
I think that `table_type != GRN_OBJ_TABLE_HASH_KEY && table_type != GRN_OBJ_TABLE_PAT_KEY` is one group.
So, I use `()`. Because I think I should use `()` to easy to read.

Reply:
No. we don't write it like that.

For example, if table is `TABLE_PAT_KEY`, `table_type & (GRN_OBJ_TABLE_HASH_KEY | GRN_OBJ_TABLE_PAT_KEY)` is `true`. However, I don't want to execute output error log, if  table is `TABLE_PAT_KEY`.

Reply:
`table_type` isn't a flag: https://github.com/groonga/groonga/blob/ca2683888109333ceb2ec75449d8186afb622ec8/include/groonga/groonga.h#L685-L689

So we can't use bit operations.